scripts/ci/check-dependencies-versions-match.js

```diff
@@ -0,0 +1,135 @@
+} else {
+	checkDependenciesMatch( expectedDependencies, packageJsons );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ * @param {Object.<String, String>} pathMappings
+ */
+function fixDependenciesVersions( expectedDependencies, packageJsons, pathMappings ) {
+	packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.forEach( packageJson => {
+			Object.entries( packageJson.dependencies )
+				.forEach( ( [ dependency, version ] ) => {
+					if ( version !== expectedDependencies[ dependency ] ) {
+						packageJson.dependencies[ dependency ] = expectedDependencies[ dependency ];
+					}
+				} );
+
+			fs.writeJsonSync( pathMappings[ packageJson.name ], packageJson, { spaces: 2 } );
+		} );
+
+	console.log( '✅ All dependencies fixed!' );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ */
+function checkDependenciesMatch( expectedDependencies, packageJsons ) {
+	const errors = packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.flatMap( packageJson => Object.entries( packageJson.dependencies )
+			.map( ( [ dependency, version ] ) => {
+				if ( version !== expectedDependencies[ dependency ] ) {
+					return getWrongVersionErrorMsg( dependency, packageJson.name, version, expectedDependencies );
+				}
+			} )
+			.filter( Boolean )
+		);
+
+	if ( errors.length ) {
+		console.error( '❌ Errors found. Run this script with `--fix` to resolve the issues automatically.' );
+		console.error( errors );
+		process.exit( 1 );
+	}
+}
+
+/**
+ * @param {String} dependency
+ * @param {String} name
+ * @param {String} version
+ * @param {Object.<String, String>} expectedDependencies
+ * @return {String}
+ */
+function getWrongVersionErrorMsg( dependency, name, version, expectedDependencies ) {
+	return `"${ dependency }" in "${ name }" in version "${ version }" should be set to "${ expectedDependencies[ dependency ] }".`;
+}
+
+/**
+ * @param {Array.<Object>} packageJsons
+ * @return {Object.<String, String>} expectedDependencies
+ */
+function getExpectedDepsVersions( packageJsons ) {
+	return packageJsons
+		.map( packageJson => packageJson.dependencies )
+		.filter( Boolean )
+		.reduce( ( expectedDependencies, dependencies ) => {
+			Object.entries( dependencies ).forEach( ( [ dependency, version ] ) => {
+				expectedDependencies[ dependency ] = getNewestVersion( version, expectedDependencies[ dependency ] );
+			} );
+
+			return expectedDependencies;
+		}, {} );
+}
+
+/**
+ * @param {String|undefined} versionA
+ * @param {String|undefined} versionB
+ * @return {String}
+ */
+function getNewestVersion( versionA, versionB ) {
+	const versionAStripped = semver.valid( semver.coerce( versionA ) ) || '0.0.0';
+	const versionBStripped = semver.valid( semver.coerce( versionB ) ) || '0.0.0';
+
+	return semver.gt( versionAStripped, versionBStripped ) ? versionAStripped : versionBStripped;
+}
```

Comment:
I don't know if it's worth adding in this PR, but I'll describe the scenario I've been thinking about.

I'm not sure if we really should just truncate the range character `^` and take the rest as the target version of the package.

Let's assume we have the `lodash-es` package in `package.json` defined like this: `"lodash-es": ^4.17.11"`. The latest version on npm is 4.17.21, which may introduce significant bug fixes or vulnerabilities fixes comparing to 4.17.11. Previously, using the range, version 4.17.21 would be installed. Now, the exact version 4.17.11 will be installed - the version that someone once (probably a long time ago) entered into `package.json`.

So I have another idea: the function `getNewestVersion()` could first get the available versions for a given package and then calculate the newest one that matches the range:

```js
const { execSync } = require( 'child_process' );
const semver = require( 'semver' );

const versions = execSync( `npm view lodash-es versions --json`, { encoding: 'utf8' } );

// Returns 4.17.21, as it is the latest version that satisfies the ^4.17.11 range.
return semver.maxSatisfying( JSON.parse( versions ), '^4.17.11' );
```

As the npm calls are expensive, the script could cache the results for each package.

Reply:
This was actually a pretty good catch. Thanks for this point! Fixed!